src/optimizer/pushdown/pushdown_cross_product.cpp

```diff
@@ -62,9 +62,21 @@ unique_ptr<LogicalOperator> FilterPushdown::PushdownCrossProduct(unique_ptr<Logi
 		                                             op->children[1], left_bindings, right_bindings, join_expressions,
 		                                             conditions, arbitrary_expressions);
 		// create the join from the join conditions
-		return LogicalComparisonJoin::CreateJoin(GetContext(), join_type, join_ref_type, std::move(op->children[0]),
-		                                         std::move(op->children[1]), std::move(conditions),
-		                                         std::move(arbitrary_expressions));
+		auto new_op = LogicalComparisonJoin::CreateJoin(GetContext(), join_type, join_ref_type,
+		                                                std::move(op->children[0]), std::move(op->children[1]),
+		                                                std::move(conditions), std::move(arbitrary_expressions));
+
+		// possible cases are: AnyJoin, ComparisonJoin, or Filter + ComparisonJoin
+		if (op->has_estimated_cardinality) {
+			// set the estimated cardinality of the new operator
+			new_op->SetEstimatedCardinality(op->estimated_cardinality);
+			if (new_op->type == LogicalOperatorType::LOGICAL_FILTER) {
```

Comment:
when would this be true? doesn't line 65 only create join operators?

Reply:
CreateJoin can return FILTER + JOIN if there are some arbitrary_expressions -- [here](https://github.com/duckdb/duckdb/blob/862852fa395b99735e5713cb55d0cea1d9320659/src/planner/binder/tableref/plan_joinref.cpp#L234)